### PR TITLE
JDK-8262944: Improve exception message when automatic module lists provider class not in JAR file

### PR DESCRIPTION
**Pull request by sormuras**
This commit appends the name of the JAR file to the exception message for when automatic module lists a non-existing provider class.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8262944](https://bugs.openjdk.java.net/browse/JDK-8262944): Improve exception message when automatic module lists provider class not in JAR file

### Reviewers
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**) ⚠️ Review applies to bb9762b890756b140a8db4ef3feaa9cb056b1607
 * [Jorn Vernee](https://openjdk.java.net/census#jvernee) (@JornVernee - **Reviewer**) ⚠️ Review applies to bb9762b890756b140a8db4ef3feaa9cb056b1607
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Mandy Chung](https://openjdk.java.net/census#mchung) (@mlchung - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5543/head:pull/5543` \
`$ git checkout pull/5543`

Update a local copy of the PR: \
`$ git checkout pull/5543` \
`$ git pull https://git.openjdk.java.net/jdk pull/5543/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5543`

View PR using the GUI difftool: \
`$ git pr show -t 5543`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5543.diff">https://git.openjdk.java.net/jdk/pull/5543.diff</a>

</details>

### CODE REVIEW

**Review by dfuch — APPROVED** [review]
The proposed change looks reasonable to me.

**Review by JornVernee — APPROVED** [review]

File: src/java.base/share/classes/jdk/internal/module/ModulePath.java [hunk 1]
```diff
@@ -551,7 +551,7 @@ private ModuleDescriptor deriveModuleDescriptor(JarFile jf)
                     if (!cn.isEmpty()) {
                         String pn = packageName(cn);
                         if (!packages.contains(pn)) {
-                            String msg = "Provider class " + cn + " not in module";
+                            String msg = "Provider class " + cn + " not in module created for " + fn;
```
Comment (AlanBateman) [inline src/java.base/share/classes/jdk/internal/module/ModulePath.java:554]: "not in module created" isn't quite right as the module cannot be created.  Maybe you could change it to "not in JAR file"

Comment (sormuras) [reply src/java.base/share/classes/jdk/internal/module/ModulePath.java:554]: Will do so.

**Review by AlanBateman — COMMENTED** [review]
Thanks for the update. For completeness, you may want to add to the list of JBS issues listed at the top of AutomatcModulesTest and you probably should update its copyright year too.

**Review by LanceAndersen — APPROVED** [review]

**Review by mlchung — APPROVED** [review]
Looks good to me.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**